arangod/Aql/ExecutionState.cpp

```diff
@@ -23,35 +23,45 @@
 
 #include "ExecutionState.h"
 
+#include "Assertions/ProdAssert.h"
+
 #include <ostream>
 
 namespace arangodb::aql {
 
-std::ostream& operator<<(std::ostream& ostream, ExecutionState state) {
+auto toStringView(ExecutionState state) -> std::string_view {
   switch (state) {
     case ExecutionState::DONE:
-      ostream << "DONE";
-      break;
+      return "DONE";
     case ExecutionState::HASMORE:
-      ostream << "HASMORE";
-      break;
+      return "HASMORE";
     case ExecutionState::WAITING:
-      ostream << "WAITING";
-      break;
+      return "WAITING";
   }
-  return ostream;
+  ADB_PROD_ASSERT(false)
+      << "Unhandled state "
+      << static_cast<std::underlying_type_t<decltype(state)>>(state);
+  std::abort();
```

Comment:
Why? Isn't assert(false) no return?

Reply:
It is, but not all compilers believe/prove that - iirc it was MSVC that complains with a warning about a missing return in a non-void function.

Reply:
Msvc needs `__assume(false)` for unreachable "tag".
Btw absl has cross platform unreachable marker

Reply:
Also it can be valid detection.
We saw that with msvc after triggered assert code continue works
(Probably out assert written incorrect)

Reply:
Anyway resolved